common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/ClasspathUtil.xtend

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.dsl.common.util
+
+import java.io.File
+import java.io.FileInputStream
+import java.io.InputStream
+import java.util.ArrayList
+import java.util.List
+import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import org.eclipse.core.runtime.IPath
+import org.eclipse.core.runtime.Path
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.xtext.EcoreUtil2
+
+class ClasspathUtil {
+
+	static val EFFECTIVE_POM_TXT_PATH = "target/effective_pom.txt"
+
+	@Inject WorkspaceRootHelper workspaceRootHelper
+
+	List<IPath> mavenClasspath
+
+	def String getPackageFromFileSystem(EObject element) {
```

Comment:
It's quite hard for me to understand what this method does as the variables are rewritten and, for example, is just named `path`. Alternate method name proposal: `inferPackage`

The approach seems reasonable, but we should definitly add some caching here - this will get really expensive otherwise.